src/Plugin/resource/Field/ResourceFieldEntity.php

```diff
@@ -421,6 +474,26 @@ public function getEntityType() {
    */
   public function setEntityType($entity_type) {
     $this->entityType = $entity_type;
+
+    // If there is no property try to get it based on the wrapper method and
+    // store the value in the decorated object.
+    $property = NULL;
+    // If entity metadata wrapper methods were used, then return the appropriate
+    // entity property.
+    if (!$this->isWrapperMethodOnEntity() || !($wrapper_method = $this->getWrapperMethod())) {
+      return NULL;
+    }
+    $entity_info = entity_get_info($entity_type);
+    if ($wrapper_method == 'label') {
```

Comment:
can you also please explain this part? I'm not sure what it does, but maybe we also need to take care of the ``$wrapper_method == 'url'`` case?